test/src/waittask.spec.ts

```diff
@@ -478,113 +479,186 @@ describe('waittask specs', function () {
       await waitForSelector;
       expect(boxFound).toBe(true);
     });
-    it('should wait for visible', async () => {
+    it('should wait for element to be visible (display)', async () => {
       const {page} = getTestState();
 
-      let divFound = false;
-      const waitForSelector = page
-        .waitForSelector('div', {visible: true})
-        .then(() => {
-          return (divFound = true);
-        });
-      await page.setContent(
-        `<div style='display: none; visibility: hidden;'>1</div>`
-      );
-      expect(divFound).toBe(false);
-      await page.evaluate(() => {
-        return document.querySelector('div')?.style.removeProperty('display');
+      const promise = page.waitForSelector('div', {visible: true});
+      await page.setContent('<div style="display: none">text</div>');
+      const element = await page.evaluateHandle(() => {
+        return document.getElementsByTagName('div')[0]!;
       });
-      expect(divFound).toBe(false);
-      await page.evaluate(() => {
-        return document
-          .querySelector('div')
-          ?.style.removeProperty('visibility');
+      await expect(
+        Promise.race([promise, createTimeout(40)])
```

Comment:
wha do we need a timeout here? generally, timeouts in tests make things more flaky, can we avoid it?

Reply:
I've written a comment about this. See above. It won't make the tests flaky because we are just assuring the default hasn't changed. If it does, then the test did its job.